src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.cs

```diff
@@ -95,6 +90,26 @@ public IEnumerable<ISymbol> Find(IAssemblySymbol assembly, Func<string, bool> pr
             }
         }
 
+        /// <summary>
+        /// Slow, linear scan of all the symbols in this assembly to look for matches.
+        /// </summary>
+        public IEnumerable<ISymbol> Find(IAssemblySymbol assembly, Func<string, bool> predicate, CancellationToken cancellationToken)
+        {
+            for (int i = 0, n = _nodes.Count; i < n; i++)
```

Comment:
It's more that i don't think of it as an optimization.  I just think of it as the way to enumerate things when you want an index.  It's idiomatic for me.  I'm not sure it warrants any additional info.